cmdr/src/giti/branch/checkout.rs

```diff
@@ -0,0 +1,149 @@
+/*
+ *   Copyright (c) 2023 R3BL LLC
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+use std::process::Command;
+
+use r3bl_ansi_color::{AnsiStyledText, Color, Style};
+use r3bl_rs_utils_core::{log_debug, CommonResult};
+use r3bl_tuify::{select_from_list_with_multi_line_header,
+                 SelectionMode,
+                 StyleSheet,
+                 LIGHT_GRAY_COLOR,
+                 SUCCESS_COLOR};
+
+use super::{get_branches, try_get_current_branch};
+use crate::giti::{report_unknown_error_and_propagate, single_select_instruction_header};
+
+pub fn try_checkout_branch() -> CommonResult<()> {
+    let default_header_style = [
+        Style::Foreground(Color::Rgb(171, 204, 242)),
```

Comment:
Would it make sense to have an enum somewhere that has a bunch of these colors.
```rust
enum DefaultColors { Color1, Color2 }
impl DefaultColors {
  pub fn as_ansi_color(&self) -> r3bl_ansi_color::Color { 
    match self {
      Color1 => Color::Rgb(171,204,242);
      // etc.
    }
  } 

  pub fn as_tui_color(&self) -> r3bl_tui::TuiColor { 
    match self {
      Color1 => TUIColor::Rgb(RgbValue(red:171,green:204,blue:242));
      // etc.
    }
  } 

}
```
Alternatively you can just have a lot of top level constants.

The enum above might go in a file called `color_constants.rs` that goes in the giti folder (for giti as an example). For edi, there would be a `color_constants.rs` file in the edi folder. For common `color_constants.rs` maybe we can place these at the src/ folder (very similar to the pattern we might use with ui_strings.rs.